single linters: Add regex linter.

jiviteshjain opened the pull request:
### Sample output:
![image](https://user-images.githubusercontent.com/43912285/87475219-00b19280-c642-11ea-9537-688263de6e94.png)

### Checks for
- Commit title against a regex
- Length of the commit title
- Length of each line in the commit body

### Note:
- `LinterLevel` should inherit `IntEnum` to allow comparisons between levels.

juped commented on git_lint_branch/linter_output.py:
```diff
@@ -1,6 +1,6 @@
+from enum import IntEnum
 
-class LinterLevel(Enum):
+class LinterLevel(IntEnum):
```
This makes sense to do. Could you break it out into a different commit in the history, though? (Doesn't need to be a whole different branch, IMO.)

jiviteshjain replied:
Done. This is the one thing that our linter can't programmatically detect :p

juped reviewed (changes requested):
Looks sensible. We'll go back and wire it to configuration once that's in.

jiviteshjain committed:
LinterLevels: Change from Enum to IntEnum.

Allows for comparisons between linter levels.

jiviteshjain committed:
single linters: Add regex linter.

Checks commit message title against a regex and the line width
of the commit title as well as the commit body.

Managing indentation in multiline strings is confusing.
